### PR TITLE
#7: add prop `wrapper` for any wrapper prop (closes #7)

### LINKED ISSUE #7: add prop `wrapper` for any wrapper prop

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Issue #7

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,9 +32,6 @@
     "url": "https://github.com/buildo/react-input-link/issues"
   },
   "homepage": "https://github.com/buildo/react-input-link",
-  "dependencies": {
-    "lodash": "^3.9.3"
-  },
```
Comment (giogonzo) [inline package.json:37]: kill the deps :tada:

File: package.json [hunk 2]
```diff
@@ -51,6 +48,7 @@
     "karma-mocha": "^0.1.10",
     "karma-webpack": "^1.5.1",
     "mocha": "^2.2.5",
+    "react": "^0.13.3",
```
Comment (giogonzo) [inline package.json:51]: why this?

Comment (FrancescoCioria) [inline package.json:51]: to run examples (it's a `dev-dependency`)

**Comment by giogonzo** [comment]
So you added "customization-through-additional props" here, when instead we decided to go with "customization-through-custom-components" for similar cases on b-r-c

Is there any rule of thumb you can extract/elaborate from this?

Was it just because this component is intended to be public? or for the relatively simple customization options?

**Comment by FrancescoCioria** [comment]
> So you added "customization-through-additional props" 

Customization through prop for the `div` (which was already done with `wrapperClassName` and `wrapperStyle`): as a wrapper is necessary and needs to be populated with computed children, It is much easier to simply expose its props instead of accepting a component, cloning it and passing it the correct children.

If you referred to the `inputProps` that was (in my opinion) one of the requirements as this component wants to be a super set of `<input />` so it should be able to behave as a normal input too (this is currently contradicted by the fact that `children` is **required**... I know :) )

> Is there any rule of thumb you can extract/elaborate from this?

I'd say: 
- If the customization is passing general props (`className`, `id`, `style`, event listeners...) this should be done by exposing the component extra props
- If the customization is more incisive (for example having a completely different component or changing custom props - like `popover`) instead of creating _ad-hoc_ solutions we should give the user the chance to completely replace the default component with their own.

@giogonzo what do you think?

**Comment by giogonzo** [comment]
makes sense, thanks for the clarification

> If you referred to the inputProps that was (in my opinion) one of the requirements as this component wants to be a super set of <input /> so it should be able to behave as a normal input too (this is currently contradicted by the fact that children is required... I know :) )

I was mainly referring to wrapper props but this makes sense as approach for a superset of input, got it

**Pull request merged**